src/navigation/navigationContext.js

```diff
@@ -26,34 +26,52 @@ export const useNavigationBlocker = (when, message) => {
   const [isBlocked, setIsBlocked] = useState(false);
   const [blockedArgs, setBlockedArgs] = useState();
 
-  const unblock = useCallback(() => {
+  /*
+   * we cache a reference to the original push method if we don't have one yet,
+   * so that we can be sure we're restoring the original when the blocker is
+   * disabled or otherwised cleaned up.
+   */
+  if (!navigator.originalPush) {
+    navigator.originalPush = navigator.push;
```

Comment:
Suggestion: Since `originalPush` is something only relevant within this code, I think it is better if we don't create a new property inside `navigator`, since it is a definition that we don't own, I think we could create a separated originalPush variable, or maybe use useRef or useState if that is needed, but it seems that a variable might work

Reply:
That's a fair point - this seemed the most straightforward way but I could try to find another. The simple approach of just using a variable in the method body here doesn't work because we'll lose the reference to the original push method after the render runs a second time, which was why I did this (since we were already doing things to the `navigator` object's members).

Reply:
I tried a few things and ended up settling in a variable for this module. Let me know if you'd like to discuss this further before merging.